projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php

```diff
@@ -1073,12 +1073,31 @@ function get_paywall_blocks( $newsletter_access_level ) {
 		? __( 'Already a paid subscriber?', 'jetpack' )
 		: __( 'Already a subscriber?', 'jetpack' );
 
-	$sign_in = '';
-	if ( ! is_user_logged_in() && ( new Host() )->is_wpcom_simple() ) {
-		$sign_in_link = wpcom_logmein_redirect_url( get_current_url(), false, null, 'link' );
+	$sign_in             = '';
+	$show_button_sign_in = false;
+	$host                = new Host();
+	if ( ! is_user_logged_in() && $host->is_wpcom_simple() ) {
+		$id                  = '';
+		$show_button_sign_in = true;
+		$sign_in_link        = wpcom_logmein_redirect_url( get_current_url(), false, null, 'link' );
+		$button_text         = esc_html__( 'Log in', 'jetpack' );
+	} elseif ( class_exists( 'Automattic\Jetpack\Connection\Tokens\Jetpack_Token_Subscription_Service' ) &&
+		! $host->is_wpcom_platform() ) {
+		// We are on Jetpack and no cookie is set
+		$show_button_sign_in = true;
+		$id                  = 'retrieve_subscriptions_link';
+		$sign_in_link        = 'javascript:show_iframe_retrieve_subscriptions_from_email()';
```

Comment:
Minor but probably best to just have `#` (or real fallback link) here and attach the click listener directly in JS after DOM finishes loading.

If the site is implementing [CSP](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP), they'll likely forbid JS in `href`.

Reply:
I actually forgot to remove it, but this is how it is implemented (via DOM click event)